File: backend/src/models/Treehole.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface ITreehole extends Document {
    content: string,
    userId: number,
    createTime: Date,
    reviewStatus: 'PENDING' | 'APPROVED' | 'REJECTED',
    deleted: boolean
}

const TreeholeSchema: Schema = new Schema({
    content: {
        type: String,
        required: [true, '弹幕内容不能为空'],
        trim: true
    },
    userId: {
        type: Number,
        required: true
    },
    createTime: {
        type: Date,
        required: [true, '发送时间不能为空'],
        trim: true
    },
    reviewStatus: {
        type: String,
        enum: ['PENDING', 'APPROVED', 'REJECTED'],
        default: 'PENDING'
    },
    deleted: {
        type: Boolean,
        default: false
    }
}, {
    timestamps: {
        createdAt: 'createTime',
        updatedAt: 'updateTime'
    }
});

const Treehole = mongoose.model<ITreehole>('treehole_messages', TreeholeSchema);
export default Treehole;

File: backend/java-backend/src/main/java/com/example/demo/model/mongo/Treehole.java
package com.example.demo.model.mongo;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document(collection = "treehole_messages")
public class Treehole {

    @Id
    private String id;

    private String content;
    private Integer userId;
    private LocalDateTime createTime;
    private String reviewStatus;  // PENDING, APPROVED, REJECTED
    private Boolean deleted;
}
